contract-examples/contracts/ExampleFeeManager.sol

```diff
@@ -84,3 +103,131 @@ contract ExampleFeeManager is AllowList {
+    assertEq(manager.readAllowList(exampleAddress), 1);
+  }
+
+  function test_changeFees() public {
+    ExampleFeeManager example = new ExampleFeeManager();
+    address exampleAddress = address(example);
+
+    IFeeManager manager = IFeeManager(FEE_MANAGER_ADDRESS);
+
+    manager.setEnabled(exampleAddress);
+
+    FeeConfig memory config = example.getCurrentFeeConfig();
+
+    FeeConfig memory newFeeConfig = FeeConfig({
+      gasLimit: CCHAIN_GAS_LIMIT,
+      targetBlockRate: CCHAIN_TARGET_BLOCK_RATE,
+      minBaseFee: CCHAIN_MIN_BASE_FEE,
+      targetGas: CCHAIN_TARGET_GAS,
+      baseFeeChangeDenominator: CCHAIN_BASE_FEE_CHANGE_DENOMINATOR,
+      minBlockGasCost: CCHAIN_MIN_BLOCK_GAS_COST,
+      maxBlockGasCost: CCHAIN_MAX_BLOCK_GAS_COST,
+      blockGasCostStep: CCHAIN_BLOCK_GAS_COST_STEP
+    });
+
+    assertNotEq(config.gasLimit, newFeeConfig.gasLimit);
+    // target block rate is the same for wagmi and cchain
+    // assertNotEq(config.targetBlockRate, newFeeConfig.targetBlockRate);
+    assertNotEq(config.minBaseFee, newFeeConfig.minBaseFee);
+    assertNotEq(config.targetGas, newFeeConfig.targetGas);
+    assertNotEq(config.baseFeeChangeDenominator, newFeeConfig.baseFeeChangeDenominator);
+    // min block gas cost is the same for wagmi and cchain
+    // assertNotEq(config.minBlockGasCost, newFeeConfig.minBlockGasCost);
+    assertNotEq(config.maxBlockGasCost, newFeeConfig.maxBlockGasCost);
+    assertNotEq(config.blockGasCostStep, newFeeConfig.blockGasCostStep);
+
+    example.enableCChainFees();
+
+    FeeConfig memory changedFeeConfig = example.getCurrentFeeConfig();
+
+    assertEq(changedFeeConfig.gasLimit, newFeeConfig.gasLimit);
+    // target block rate is the same for wagmi and cchain
+    // assertEq(changedFeeConfig.targetBlockRate, newFeeConfig.targetBlockRate);
+    assertEq(changedFeeConfig.minBaseFee, newFeeConfig.minBaseFee);
+    assertEq(changedFeeConfig.targetGas, newFeeConfig.targetGas);
+    assertEq(changedFeeConfig.baseFeeChangeDenominator, newFeeConfig.baseFeeChangeDenominator);
+    // min block gas cost is the same for wagmi and cchain
+    // assertEq(changedFeeConfig.minBlockGasCost, newFeeConfig.minBlockGasCost);
+    assertEq(changedFeeConfig.maxBlockGasCost, newFeeConfig.maxBlockGasCost);
+    assertEq(changedFeeConfig.blockGasCostStep, newFeeConfig.blockGasCostStep);
+
+    assertEq(example.getFeeConfigLastChangedAt(), block.number);
+
+    example.enableCustomFees(config);
+  }
+
+  function test_minFeeTransaction() public {
+    // noop
+  }
+
+  function test_raiseMinFeeByOne() public {
+    ExampleFeeManager example = new ExampleFeeManager();
+    address exampleAddress = address(example);
+
+    IFeeManager manager = IFeeManager(FEE_MANAGER_ADDRESS);
+
+    manager.setEnabled(exampleAddress);
+
+    FeeConfig memory config = example.getCurrentFeeConfig();
+    config.minBaseFee = config.minBaseFee + 1;
+
+    example.enableCustomFees(config);
```

Comment:
if these functions are used for before/after tests can we specify them in the name like test_beforeXXX/test_afterXXX where XXX is the test name

Reply:
We don't need to enforce these names but just use them as the convention. IMO this would make it much easier to read and not mix them with actual tests.

Reply:
But you can't "not mix them with the actual tests,"... though that would be nice. 

A `test_*` function itself does not constitute a test, but it should be run in a testing context. The `test_` prefix is enforced at the call site though there's nothing preventing a `test_` function from being called outside of a test. 

The issue with adding `test_beforeMyTest` and `test_afterMyTest` is that it implies `test_myTest` is a test, which actually isn't the case. You *must* read the TypeScript code in order to fully understand what is being tested, and when you do, the order in which the `test_*` methods are invoked is clear. *Not fully understanding* what's being tested in `*.sol` files is essentially a feature. 

While this is far from ideal (if only we could plug `subnet-evm` directly into `Forge`), it's actually no worse than what was previously here. While understanding a test *still* requires context in both Solidity and TypeScript, the vast majority of that context is now in Solidity, which IMO is an improvement.

Reply:
I get these cannot be separated from the test framework, however this is not clearly understandable from this file. As I said on my previous comments, these contracts also meant to be a documentation/tutorial for precompiles; so these should be clear/clean as possible. I'm fine with either:
1- adding comments that refers to the actual test in TS file and for full context reader should be looking at there as well.  
2- Or adding comments to functions/renaming function names in .sol file, explaining them in the .sol file.

I still strongly think we should move those test parts to another file under `contracts/tests/ExampleFeeManagerTest.sol` etc.

Reply:
> I still strongly think we should move those test parts to another file under contracts/tests/ExampleFeeManagerTest.sol etc.

This seem preferable to me as well.

Should we refer to these functions as `testStep_` or `testTx_` or `step_` instead of `test_`? 

`test_` implies that this function is a full test in most testing frameworks.

Reply:
@ceyonur and @darioush, please see my comment here:
https://github.com/ava-labs/subnet-evm/pull/601#discussion_r1181830561

and feel free to continue the discussion in that thread. 

If you guys _**don't agree**_ with a separate repo with the examples, then I will move the tests to `contract/tests/*.sol`

Reply:
Update: moved everything to `contract/test/*.sol`